tests/unit/apps/test_base.py

```diff
@@ -281,3 +281,69 @@ def test_get_reached_expected_target_step(mock_workload_upgrade, units, model):
 
     app._get_reached_expected_target_step(target, units)
     mock_workload_upgrade.assert_has_calls(expected_calls)
+
+
```

Comment:
Maybe it worth parameterize the config and have the following scenarios:

1. no config for `enable-auto-restarts` (already implemented)
2. `enable-auto-restarts` set to True